packages/mdc-list/foundation.js

```diff
@@ -371,6 +412,38 @@ class MDCListFoundation extends MDCFoundation {
   hasCheckboxOrRadioAtIndex_(index) {
     return this.adapter_.hasCheckboxAtIndex(index) || this.adapter_.hasRadioAtIndex(index);
   }
+
+  /**
+   * @param {!Index} index
+   * @return {boolean}
+   */
+  isIndexValid_(index) {
+    const listSize = this.adapter_.getListItemCount();
+
+    if (index instanceof Array) {
+      if (!this.hasCheckboxAtIndex_(index)) {
+        throw new Error('MDCListFoundation: Array of index is only supported for checkbox based list');
+      }
+      return index.some((i) => this.isIndexInRange_(i));
```

Comment:
Should this be .every instead of .some? What happens if one index is in range but another isn't?

Reply:
I removed the error throwing when index is not in range, so it should ignore them when out of range and select rest.